### PR TITLE
Update karma and easy deps

### PR DESCRIPTION
**Pull request by alexdlm**
## Before this PR

I couldn't `yarn install` on my environment due to `karma -> chokidar -> fsevents` old deps that wouldn't compile, whether that's OSX Monterey or some other environment thing not sure but seems sensible to update anyway. Should unblock the karma updating excavator for the future

## After this PR

==COMMIT_MSG==
Update karma and various deps.
==COMMIT_MSG==

## Possible downsides?
Tested on node 12.19, 12.20, 14.20 and 16.16
I didn't touch other deps (such as yargs which is many major versions behind) that require other code changes...

### CODE REVIEW

File: src/commands/generate/__tests__/cliTest.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ describe("generate command", () => {
     let outDir: string;
     const input = path.join(__dirname, "../../../../build/ir-test-cases/example-types.json");
     const generateCommand = new GenerateCommand();
-    jest.setTimeout(10000);
+    jest.setTimeout(30000);
```
Comment (alexdlm) [inline src/commands/generate/__tests__/cliTest.ts:29]: Maybe my slow internet or struggling machine I was flaking on this

File: src/commands/generate/generateCommand.ts [hunk 2]
```diff
@@ -146,7 +146,7 @@ export class GenerateCommand implements CommandModule {
             throw Error('Must either specify "rawSource" or specify "packageName" and "packageVersion"');
         } else if (!fs.existsSync(output)) {
             throw new Error(`Directory "${output}" does not exist`);
-        } else if (!rawSource && !isValid(packageVersion!)) {
+        } else if (!rawSource && !SlsVersion.isValid(packageVersion!)) {
```
Comment (alexdlm) [inline src/commands/generate/generateCommand.ts:149]: Broke my own no-code-change threshold, but this seemed reasonable & simple for a palantir dep

File: package.json [hunk 3]
```diff
@@ -28,50 +28,48 @@
...
+    "semver": "^7.3.7",
+    "sls-version": "^2.1.0",
     "ts-simple-ast": "^11.3.0",
     "tslib": "^1.9.3",
     "yargs": "^11.1.0"
   },
   "devDependencies": {
     "@blueprintjs/tslint-config": "^1.6.0",
     "@types/chai": "^4.1.6",
-    "@types/commander": "2.12.2",
```
Comment (alexdlm) [inline package.json:44]: `commander` included its own typings at some point after this, and this `@types/commander` package had a dependency on `commander: "*"` which brought in `commander@9.something`, even though it wasn't used, but it wouldn't install on node < 12.20.

File: package.json [hunk 4]
```diff
@@ -28,50 +28,48 @@
...
-    "@types/download": "^6.2.2",
```
Comment (alexdlm) [inline package.json:45]: wasn't used

File: src/commands/generate/__tests__/serviceGeneratorTest.ts [hunk 5]
```diff
@@ -517,7 +517,7 @@ export interface IMyService {
 
         expect(contents).not.toContain(
             `
-            /** 
+            /**
```
Comment (alexdlm) [inline src/commands/generate/__tests__/serviceGeneratorTest.ts:520]: tslint update caught this trailing space, which highlights that negative tests aren't great.

**Review by p-szm — APPROVED** [review]

**Pull request merged**